packages/calcite-components/src/components/carousel/carousel.e2e.ts

```diff
@@ -0,0 +1,804 @@
+      expect(selectedItem.id).toEqual("three");
+
+      await page.keyboard.press("ArrowRight");
+      await page.waitForChanges();
+      expect(changeSpy).toHaveReceivedEventTimes(6);
+      selectedItem = await carousel.find(`calcite-carousel-item[selected]`);
+      expect(selectedItem.id).toEqual("one");
+    });
+
+    it("emit event on navigation via direct pagination item click", async () => {
+      const page = await newE2EPage();
+
+      await page.setContent(
+        `<calcite-carousel label="Carousel example" arrow-type="none">
+          <calcite-carousel-item label="Slide 1" id="one"><p>no pre-selected attribute</p></calcite-carousel-item>
+          <calcite-carousel-item label="Slide 2" id="two" selected><p>pre-selected and not first</p></calcite-carousel-item>
+          <calcite-carousel-item label="Slide 3" id="three"><p>no pre-selected attribute</p></calcite-carousel-item>
+        </calcite-carousel>`,
+      );
+
+      const carousel = await page.find("calcite-carousel");
+      const changeSpy = await page.spyOnEvent("calciteCarouselChange");
+      const slide1Button = await page.find(`calcite-carousel >>> .${CSS.paginationItemIndividual}:nth-child(1)`);
+      const slide2Button = await page.find(`calcite-carousel >>> .${CSS.paginationItemIndividual}:nth-child(2)`);
+      const slide3Button = await page.find(`calcite-carousel >>> .${CSS.paginationItemIndividual}:nth-child(3)`);
+
+      expect(changeSpy).not.toHaveReceivedEvent();
+      let selectedItem = await carousel.find(`calcite-carousel-item[selected]`);
+      expect(selectedItem.id).toEqual("two");
+
+      await slide1Button.click();
+      await page.waitForChanges();
+      expect(changeSpy).toHaveReceivedEventTimes(1);
+      selectedItem = await carousel.find(`calcite-carousel-item[selected]`);
+      expect(selectedItem.id).toEqual("one");
+
+      await slide1Button.click();
+      await page.waitForChanges();
+      expect(changeSpy).toHaveReceivedEventTimes(1);
+      selectedItem = await carousel.find(`calcite-carousel-item[selected]`);
+      expect(selectedItem.id).toEqual("one");
+
+      await slide3Button.click();
+      await page.waitForChanges();
+      expect(changeSpy).toHaveReceivedEventTimes(2);
+      selectedItem = await carousel.find(`calcite-carousel-item[selected]`);
+      expect(selectedItem.id).toEqual("three");
+
+      await slide2Button.click();
+      await page.waitForChanges();
+      expect(changeSpy).toHaveReceivedEventTimes(3);
+      selectedItem = await carousel.find(`calcite-carousel-item[selected]`);
+      expect(selectedItem.id).toEqual("two");
+
+      await slide2Button.click();
+      await page.waitForChanges();
+      expect(changeSpy).toHaveReceivedEventTimes(3);
+      selectedItem = await carousel.find(`calcite-carousel-item[selected]`);
+      expect(selectedItem.id).toEqual("two");
+    });
+
+    it("emit autoplay change event but not carousel change on autoplay button click", async () => {
```

Comment:
Nitpick: might be helpful to have all autoplay related tests under an autoplay describe block.